### ISSUE TITLE
print_svg_path MethodError

### ISSUE THREAD

**Issue by jiahao** [dcjones/Compose.jl#113]
My [parallel prefix notebook](https://github.com/jiahao/ijulia-notebooks/blob/master/2014-08-06-parallel-prefix.ipynb) no longer works on master :(

Reduced test case:

```jl
julia> draw(SVG(3inch,3inch), compose(context(),
        compose(context(), stroke("black"), fill("white"),
            circle(.5,.5, 0.5)),
        compose(context(), linewidth(0.3mm), stroke("black"),
            line([(0,0),(1,1)]))))
`print_svg_path` has no method matching print_svg_path(::IOBuffer, ::Array{Point{XM<:Measure{S,T},YM<:Measure{S,T}},1}, ::Bool)

 in draw at /Users/jiahao/.julia/v0.3/Compose/src/svg.jl:852
 in draw at /Users/jiahao/.julia/v0.3/Compose/src/svg.jl:754
 in draw at /Users/jiahao/.julia/v0.3/Compose/src/form.jl:23
 in drawpart at /Users/jiahao/.julia/v0.3/Compose/src/container.jl:510
 in draw at /Users/jiahao/.julia/v0.3/Compose/src/container.jl:400
```

PNG rendering works just fine.

**Comment by dcjones** [comment]
I'm struggling to reproduce this. Are you sure this is on master?

**Comment by jiahao** [comment]
Yes.

I experimented with the code and the failure mode is rather bizarre. 

```jl
using Compose

#If uncommented, both draws will succeed. If comment, the draw below will fail.
#draw(SVG(3inch,3inch), compose(context(),
#        compose(context(), stroke("black"), fill("white"),
#            circle(.5,.5, 0.5)),
#        compose(context(), linewidth(0.3mm), stroke("black"),
#            line([(0,0),(1,1)]))))

type gate
    ins :: Vector
    outs:: Vector
end

function dr(G::gate, x₁, y₁, y₀; rᵢ=0.1, rₒ=0.25)
    ipoints = [(i, y₀+rᵢ) for i in G.ins]
    opoints = [(i, y₀+0.5) for i in G.outs]
    igates  = [circle(i..., rᵢ) for i in ipoints]
    ogates  = [circle(i..., rₒ) for i in opoints]
    lines = [line([i, j]) for i in ipoints, j in opoints]
    compose(context(units=UnitBox(0.5,0,x₁,y₁+1)),
        compose(context(), stroke("black"), fill("white"),
            igates..., ogates...),
        compose(context(), linewidth(0.3mm), stroke("black"),
            lines...))
end

A=gate({1,2},{2})
draw(SVG(3inch,3inch), dr(A,2,0,0))
```

**Comment by dcjones** [comment]
Julia 0.3, 0.4, both?

**Comment by jiahao** [comment]
I saw this on 0.3.6.

**Comment by dcjones** [comment]
I can reproduce this now, but it's no less perplexing.

If I replace
```julia
print_svg_path(img.out, prim.points, true)
```
with
```julia
f = @which print_svg_path(img.out, prim.points, true)
f.func(img.out, prim.points, true)
```
It suddenly works. So this seems a lot like a julia method dispatch bug. It works without modification on julia 0.4.

I found I can work around this by relaxing the type annotations for `print_svg_path` though.

**Comment by dcjones** [comment]
The last time I saw an issue like this, it was https://github.com/JuliaLang/julia/issues/265, but I don't see any way in which it's applicable here.

**Comment by jiahao** [comment]
Thanks! I guess this is some weird thing that got fixed as part of JuliaLang/julia#8974 and @JeffBezanson's recent improvements to subtyping.
